project/AkkaBuild.scala

```diff
@@ -137,12 +137,23 @@ object AkkaBuild {
     // default JVM config for tests
     javaOptions in Test ++= {
       val defaults = Seq(
-        // memory
+        // ## core memory settings
         "-XX:+UseG1GC",
         "-Xms2g", "-Xmx2g",
-        "-XX:+AlwaysPreTouch", "-XX:-UseBiasedLocking", "-XX:+UseCompressedOops",
-
-        "-XX:MaxDirectMemorySize=256m", "-Xss2m",
+        // increate stack size (todo why?)
+        "-Xss2m",
+
+        // ## extra memory/gc tuning
+        // make sure we actually allocate physical memory (todo is it really worth it?)
+        "-XX:+AlwaysPreTouch",
+        // not sure why we disable this (todo motivate)
+        "-XX:-UseBiasedLocking",
+        // this breaks jstat, but could avoid costly syncs to disc see http://www.evanjones.ca/jvm-mmap-pause.html
+        "-XX:+PerfDisableSharedMem",
+        // tell G1GC that we would be really happy if all GC pauses could be kept low and not cause test timeouts
```

Comment:
`G1GC` shines AFAIK after `4g` heap size, by default the target is `200ms`
With `2g` heap size I'd recommend having this set to `100ms` to have the wanted effect.

Reply:
600ms is what we have been running with for the multimode tests, we'll see what the gc logs say after validation has run.

Reply:
Heh, I misremembered just copy-pasta-ing that 600 in there, but I did set it to 200

Reply:
`-XX:+ParallelRefProcEnabled` should have bigger impact than `-XX:MaxGCPauseMillis=200` for tests that generates lots of messages.

Reply:
Why, we hardly use weak refs?

Reply:
It is not only used for weak references, at least that's what I understand from here:
https://docs.oracle.com/javase/9/gctuning/garbage-first-garbage-collector-tuning.htm#GUID-40B64CD4-9844-4E3E-A0BB-81556AC04C74